server/services/professionalContactsService.ts

```diff
@@ -243,12 +251,58 @@ export default class ProfessionalContactsService {
     bookingId: number,
     prisonerNumber: string,
     prisonId: string,
+    isNewKeyWorkerServiceEnabled: boolean,
     apiErrorCallback: (error: Error) => void = () => null,
   ): Promise<StaffContacts> {
     const prisonApi = this.prisonApiClientBuilder(clientToken)
     const prisonerProfileDeliusApiClient = this.prisonerProfileDeliusApiClientBuilder(clientToken)
     const allocationManagerApiClient = this.allocationApiClientBuilder(clientToken)
 
+    if (isNewKeyWorkerServiceEnabled) {
+      const keyWorkerClient = this.keyworkerApiClientBuilder(clientToken)
+
+      const [communityManager, allocationManager, currentAllocatedStaff, bookingContacts] = await Promise.all([
+        Result.wrap(prisonerProfileDeliusApiClient.getCommunityManager(prisonerNumber), apiErrorCallback),
+        Result.wrap(allocationManagerApiClient.getPomByOffenderNo(prisonerNumber), apiErrorCallback),
+        Result.wrap(keyWorkerClient.getCurrentAllocations(prisonerNumber), apiErrorCallback),
+        prisonApi.getBookingContacts(bookingId),
+      ])
+
+      return {
+        keyWorker: currentAllocatedStaff
+          .map(({ allocations, latestRecordedEvents, hasHighComplexityOfNeeds }) => {
+            const allocatedKeyWorker = allocations.find(itm => itm.policy.code === 'KEY_WORKER')?.staffMember
+            const lastKeyWorkerSession = latestRecordedEvents.find(itm => itm.type === 'SESSION')
```

Comment:
I don't think `latestRecordedEvents` can be null but might be worth adding `?` anyway for extra prisoner profile safety. Also theoretically possible there could be more than one `SESSION`. Maybe add a first or default equivalent?